### ISSUE TITLE
Bring server up to date with schemas

### ISSUE THREAD

**Issue by macieksmuga** [ga4gh/server#599]
Schemas change: https://github.com/ga4gh/schemas/pull/378 needs to be reflected in the server.

**Comment by hjellinek** [comment]
I updated the Compliance suite to match the schemas, including https://github.com/ga4gh/schemas/pull/378.  Now several tests fail with HTTP response `404` in calls to `searchReferences` with default `SearchReferencesRequest`.  The exception body is

```
{"errorCode":1854344246,"message":"referenceSetId 'None' not found"}
```
(`errorCode` may vary)

I'm doing this to get the reference ID.  The `client.references.searchReferences(refsReq)` call throws the exception.

```java
    public static String getValidReferenceId(Client client) throws AvroRemoteException {
        final SearchReferencesRequest refsReq = SearchReferencesRequest.newBuilder().build();
        final SearchReferencesResponse refsResp = client.references.searchReferences(refsReq);
        final List<Reference> references = refsResp.getReferences();

        return references.get(0).getId();
    }
```

**Comment by hjellinek** [comment]
This may be a thinko on my part, or it may be a different bug entirely.  According to the comments in `SearchReferencesRequest` in `referencemethods.avdl`,

```
/**
  If not null, return only references which belong to this reference set.
  */
  union { null, string } referenceSetId = null;
```
it should be legit to issue a `SearchReferencesRequest` without settings its `referenceSetId` field.

However, when I do set it, the call to `searchReferences` works.  The modified code is:

```java
public static String getValidReferenceId(Client client) throws AvroRemoteException {
        final SearchReferenceSetsRequest refSetsReq = SearchReferenceSetsRequest.newBuilder().build();
        final SearchReferenceSetsResponse refSetsResp = client.references.searchReferenceSets(refSetsReq);

        final List<ReferenceSet> refSets = refSetsResp.getReferenceSets();
        final SearchReferencesRequest refsReq = SearchReferencesRequest
                .newBuilder()
                .setReferenceSetId(refSets.get(0).getId())
                .build();
        final SearchReferencesResponse refsResp = client.references.searchReferences(refsReq);
        final List<Reference> references = refsResp.getReferences();

        return references.get(0).getId();
    }
```

**Comment by jeromekelleher** [comment]
See PR ga4gh/schemas#392 for an update on this. Basically, this is a bug in the schemas in my view and should be fixed.

**Comment by jeromekelleher** [comment]
Looks like @dcolligan is working on this in #600.

**Comment by jeromekelleher** [comment]
There is one more PR open on the schemas that we need to include before we can close this issue: ga4gh/schemas#403.

**Comment by dcolligan** [comment]
Mostly fixed in #600

**Comment by jeromekelleher** [comment]
Fixed in #623.
